libs/s25main/mapGenerator/RandomMap.cpp

```diff
@@ -98,6 +97,28 @@ namespace rttr { namespace mapGenerator {
         return 6;
     }
 
+    unsigned GetIslandSize(const MapExtent& size)
+    {
+        const unsigned combinedSize = size.x + size.y;
+        if(combinedSize <= 128)
```

Comment:
I think it makes more sense to use the number of nodes of the map for deciding (I.e. `prodOfComponents(size)`)
Because a 100 * 28 map and a 64*64 differ by a factor of 2 in nodes.

Also minor style: How about using `if-else if-else` chain without braces? That would move the cases more together and although the else is technically redundant I think it reads easier here as you don't need to read the indented line to understand the structure.